### ISSUE TITLE
What does frame_dir mean?

### ISSUE THREAD

**Issue by yuchen70** [zugexiaodui/campus_vad_code#13]
Dear author，I want to follow your work, but I dont know how to deal data.
What does frame_dir mean in campus_vad_code-main/main/1-train_sce.sh?
how can i get it? 
If you can, can you give me your contact information and help? I appreciate your quick response.pls
thx.

**Comment by zugexiaodui** [comment]
Sorry for the late reply.
`frame_dir` is the folder containing the extracted frames.
The structure of the directory is as follows.
```
frame_dir: /Train, /Test
- /Train: /video_name_1, /video_name_2 ....
-- /video_name_1: / 0000.jpg, 0001.jpg ...
... ...
```

**Comment by yinyexingchen** [comment]
> Sorry for the late reply. `frame_dir` is the folder containing the extracted frames. The structure of the directory is as follows.
> 
> ```
> frame_dir: /Train, /Test
> - /Train: /video_name_1, /video_name_2 ....
> -- /video_name_1: / 0000.jpg, 0001.jpg ...
> ... ...
> ```
Dear author, thank you for your great work, I have a few questions and need your help.
1.What is ‘video_dir’? There is already ‘frame_dir’, so what is the path video_dir used for?
2.Does background information also need to be extracted in advance? The tracking part is not reflected in the paper, so I have this question.
3.Can gt.npz be generated from scratch using make_new_gt in the tool file?
4.Can the computing power used by the entire project be satisfied by just one 4090. I see the traning sh just use one.

**Comment by yinyexingchen** [comment]
> Dear author，I want to follow your work, but I dont know how to deal data. What does frame_dir mean in campus_vad_code-main/main/1-train_sce.sh? how can i get it? If you can, can you give me your contact information and help? I appreciate your quick response.pls thx.

Hello bro, the video dir seems like nnecessary. There is no video in test of Shanghai dataset. Have you solved this? Thanks in advance!

**Comment by yuchen70** [comment]
> > Dear author，I want to follow your work, but I dont know how to deal data. What does frame_dir mean in campus_vad_code-main/main/1-train_sce.sh? how can i get it? If you can, can you give me your contact information and help? I appreciate your quick response.pls thx.
> 
> Hello bro, the video dir seems like nnecessary. There is no video in test of Shanghai dataset. Have you solved this? Thanks in advance!

It's been a while since I reproduced this paper, and I can't remember some details clearly.
Frame_dir is used to store the split video frames. For example, if a video has 100 frames, then the corresponding frame_dir stores the 100 split images of the video. This task is very simple and can be implemented by yourself.
Whether a video must be based on the method you want to use, if your method requires extracting object level images such as people, bicycles, etc., then object tracking algorithms need to be used for preprocessing, and some jobs may provide this data. If your method only uses frame level information, then video is not necessary.

**Comment by yinyexingchen** [comment]
> > > Dear author，I want to follow your work, but I dont know how to deal data. What does frame_dir mean in campus_vad_code-main/main/1-train_sce.sh? how can i get it? If you can, can you give me your contact information and help? I appreciate your quick response.pls thx.
> > 
> > 
> > Hello bro, the video dir seems like nnecessary. There is no video in test of Shanghai dataset. Have you solved this? Thanks in advance!
> 
> It's been a while since I reproduced this paper, and I can't remember some details clearly. Frame_dir is used to store the split video frames. For example, if a video has 100 frames, then the corresponding frame_dir stores the 100 split images of the video. This task is very simple and can be implemented by yourself. Whether a video must be based on the method you want to use, if your method requires extracting object level images such as people, bicycles, etc., then object tracking algorithms need to be used for preprocessing, and some jobs may provide this data. If your method only uses frame level information, then video is not necessary.

Thanks for your reply bro. The author provided the tracking files,.And  is it possible to get the same result using video  the frames to track object just like hf2vad?

**Comment by zugexiaodui** [comment]
> > > > Dear author，I want to follow your work, but I dont know how to deal data. What does frame_dir mean in campus_vad_code-main/main/1-train_sce.sh? how can i get it? If you can, can you give me your contact information and help? I appreciate your quick response.pls thx.
> > > 
> > > 
> > > Hello bro, the video dir seems like nnecessary. There is no video in test of Shanghai dataset. Have you solved this? Thanks in advance!
> > 
> > 
> > It's been a while since I reproduced this paper, and I can't remember some details clearly. Frame_dir is used to store the split video frames. For example, if a video has 100 frames, then the corresponding frame_dir stores the 100 split images of the video. This task is very simple and can be implemented by yourself. Whether a video must be based on the method you want to use, if your method requires extracting object level images such as people, bicycles, etc., then object tracking algorithms need to be used for preprocessing, and some jobs may provide this data. If your method only uses frame level information, then video is not necessary.
> 
> Thanks for your reply bro. The author provided the tracking files,.And is it possible to get the same result using video the frames to track object just like hf2vad?

The tracking results were obtained by ByteTrack with post-process. Using other algs cannot reproduce the same tracking objects.

**Comment by yinyexingchen** [comment]
> > > > > Dear author，I want to follow your work, but I dont know how to deal data. What does frame_dir mean in campus_vad_code-main/main/1-train_sce.sh? how can i get it? If you can, can you give me your contact information and help? I appreciate your quick response.pls thx.
> > > > 
> > > > 
> > > > Hello bro, the video dir seems like nnecessary. There is no video in test of Shanghai dataset. Have you solved this? Thanks in advance!
> > > 
> > > 
> > > It's been a while since I reproduced this paper, and I can't remember some details clearly. Frame_dir is used to store the split video frames. For example, if a video has 100 frames, then the corresponding frame_dir stores the 100 split images of the video. This task is very simple and can be implemented by yourself. Whether a video must be based on the method you want to use, if your method requires extracting object level images such as people, bicycles, etc., then object tracking algorithms need to be used for preprocessing, and some jobs may provide this data. If your method only uses frame level information, then video is not necessary.
> > 
> > 
> > Thanks for your reply bro. The author provided the tracking files,.And is it possible to get the same result using video the frames to track object just like hf2vad?
> 
> The tracking results were obtained by ByteTrack with post-process. Using other algs cannot reproduce the same tracking objects.

Thanks for your reply！Do the npz files of the Shanghai dataset need to be re-spliced ​​with test_frame_mask, and do I need to re-convert the video frames of the test set into videos?You know these files of shanghai are not as complete as NWPU.By the way, I notice that only train_sce.sh and test_ft.sh use the frame_dir parameter  in the source code. Is this normal?

**Comment by zugexiaodui** [comment]
> > > > > > Dear author，I want to follow your work, but I dont know how to deal data. What does frame_dir mean in campus_vad_code-main/main/1-train_sce.sh? how can i get it? If you can, can you give me your contact information and help? I appreciate your quick response.pls thx.
> > > > > 
> > > > > 
> > > > > Hello bro, the video dir seems like nnecessary. There is no video in test of Shanghai dataset. Have you solved this? Thanks in advance!
> > > > 
> > > > 
> > > > It's been a while since I reproduced this paper, and I can't remember some details clearly. Frame_dir is used to store the split video frames. For example, if a video has 100 frames, then the corresponding frame_dir stores the 100 split images of the video. This task is very simple and can be implemented by yourself. Whether a video must be based on the method you want to use, if your method requires extracting object level images such as people, bicycles, etc., then object tracking algorithms need to be used for preprocessing, and some jobs may provide this data. If your method only uses frame level information, then video is not necessary.
> > > 
> > > 
> > > Thanks for your reply bro. The author provided the tracking files,.And is it possible to get the same result using video the frames to track object just like hf2vad?
> > 
> > 
> > The tracking results were obtained by ByteTrack with post-process. Using other algs cannot reproduce the same tracking objects.
> 
> Thanks for your reply！Do the npz files of the Shanghai dataset need to be re-spliced ​​with test_frame_mask, and do I need to re-convert the video frames of the test set into videos?You know these files of shanghai are not as complete as NWPU.By the way, I notice that only train_sce.sh and test_ft.sh use the frame_dir parameter in the source code. Is this normal?

1. Yes, the test frames of Shanghaitech should be converted into videos.
2. If `frame_dir` is provided, the dataloader will prioritize loading frames to avoid the slow decoding speed caused by reading the original videos. Otherwise, it will fall back to reading and decoding from the videos directly.
